sdks/python/apache_beam/io/iobase.py

```diff
@@ -1226,17 +1226,13 @@ def _finalize_write(
 
 
 class _RoundRobinKeyFn(core.DoFn):
-  def __init__(self, count):
-    # type: (int) -> None
-    self.count = count
-
   def start_bundle(self):
-    self.counter = random.randint(0, self.count - 1)
+    self.counter = random.random()
 
-  def process(self, element):
-    self.counter += 1
-    if self.counter >= self.count:
-      self.counter -= self.count
+  def process(self, element, count):
+    if self.counter < 1:
```

Comment:
I believe this was written to initialize self.counter, but it will also be true every time `(1 + self.counter) % count` is zero. Instead, you could initialize `self.counter = None` in start_bundle and check for None on this line.

Reply:
That is true, however either self.counter will either be a value in the semi-closed interval [0,1) or zero so the effect of the if statement may only be non-zero after the first start_bundle. Coding wise your suggestion is probably cleaner.